From: Janosch Frank <frankja@linux.vnet.ibm.com>
Subject: KVM: s390: Introduce BCD Vector Instructions to the guest
Patch-mainline: v4.11-rc1
Git-commit: 2f87d942be9d0f86e44fbcbd473264c26c7f1809
References: FATE#324072, LTC#158953

Summary:     kernel: Introduce enhanced vector and bcd vector facility to KVM
Description: IBM z14 introduced two new vector facilities. One brings
             new general vector instructions and one introduces BCD
             vector instructions. These two patches make them
             available for KVM guests.

Upstream-Description:

             KVM: s390: Introduce BCD Vector Instructions to the guest

             We can directly forward the vector BCD instructions to the guest
             if available and VX is requested by user space.

             Please note that user space will have to take care of the final state
             of the facility bit when migrating to older machines.

             Signed-off-by: Guenther Hutzl <hutzl@linux.vnet.ibm.com>
             Reviewed-by: Christian Borntraeger <borntraeger@de.ibm.com>
             Reviewed-by: David Hildenbrand <dahi@linux.vnet.ibm.com>
             Signed-off-by: Christian Borntraeger <borntraeger@de.ibm.com>


Signed-off-by: Janosch Frank <frankja@linux.vnet.ibm.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 arch/s390/kvm/kvm-s390.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/s390/kvm/kvm-s390.c
+++ b/arch/s390/kvm/kvm-s390.c
@@ -352,6 +352,10 @@ static int kvm_vm_ioctl_enable_cap(struc
 		} else if (MACHINE_HAS_VX) {
 			set_kvm_facility(kvm->arch.model.fac->mask, 129);
 			set_kvm_facility(kvm->arch.model.fac->list, 129);
+			if (test_facility(134)) {
+				set_kvm_facility(kvm->arch.model.fac->mask, 134);
+				set_kvm_facility(kvm->arch.model.fac->list, 134);
+			}
 			if (test_facility(135)) {
 				set_kvm_facility(kvm->arch.model.fac->mask, 135);
 				set_kvm_facility(kvm->arch.model.fac->list, 135);
